lib/rdp/headers/neg_req_header.rb

```diff
@@ -1,31 +1,31 @@
 module RDP
   class NegReqHeader
     attr_accessor :type, :flags, :length, :requested_protocol
-    
+
     def initialize(data)
       @type               = data[0]
       @flags              = data[1]
       @length             = data[2]
       @requested_protocol = data[3]
     end
-    
+
     def proto_s
       str = {
-          RDP::PROTOCOL_RDP        => 'RDP',
-          RDP::PROTOCOL_SSL        => 'SSL/TLS',
-          RDP::PROTOCOL_HYBRID     => 'HYBRID',
-          RDP::PROTOCOL_HYBRID_EXT => 'HYBRID Extended',
+        RDP::PROTOCOL_RDP        => 'RDP',
+        RDP::PROTOCOL_SSL        => 'SSL/TLS',
+        RDP::PROTOCOL_HYBRID     => 'HYBRID',
+        RDP::PROTOCOL_HYBRID_EXT => 'HYBRID Extended',
       }
-      
+
       result = []
-      result << str[RDP::PROTOCOL_RDP]        if @requested_protocol & RDP::PROTOCOL_RDP        == RDP::PROTOCOL_RDP
-      result << str[RDP::PROTOCOL_SSL]        if @requested_protocol & RDP::PROTOCOL_SSL        == RDP::PROTOCOL_SSL
-      result << str[RDP::PROTOCOL_HYBRID]     if @requested_protocol & RDP::PROTOCOL_HYBRID     == RDP::PROTOCOL_HYBRID
-      result << str[RDP::PROTOCOL_HYBRID_EXT] if @requested_protocol & RDP::PROTOCOL_HYBRID_EXT == RDP::PROTOCOL_HYBRID_EXT
-      
+      result << str[RDP::PROTOCOL_RDP]        if @requested_protocol & RDP::PROTOCOL_RDP != 0
```

Comment:
I'm not sure why you have changed it.

If a I have (1 | 2) and check to see if the result (3) contains 1, will returns 1, and if I check for 2, it will returns 2, so I'm checking to see if the result is the same as what I'm looking for instead of just if it exists.

It helps us also to know if we have a bug, that is if the result is not 0 but it's not the constant we are looking for, then something is wrong, no?